Packages/com.llealloo.audiolink/Runtime/Scripts/AudioReactiveBlendshapes.cs

```diff
@@ -0,0 +1,47 @@
+﻿using UnityEngine;
+
+namespace AudioLink
+{
+#if UDONSHARP
+    using UdonSharp;
+    using UnityEngine.Rendering.PostProcessing;
+
+    public class AudioReactiveBlendshapes : UdonSharpBehaviour
+#else
+    public class AudioReactiveBlendshapes : MonoBehaviour
+#endif
+    {
+        public AudioLink audioLink;
+        public int band;
+        [Range(0, 127)]
+        public int delay;
+        public int[] blendshapeIDs;
+        public float[] blendshapeWeights;
+        public float[] blendshapeMinWeights;
+
+        private SkinnedMeshRenderer _skinnedMeshRenderer;
+        private int _dataIndex;
+        private int _maxBlendshapes;
+
+        void Start()
+        {
+            _skinnedMeshRenderer = transform.GetComponent<SkinnedMeshRenderer>();
+            _dataIndex = (band * 128) + delay;
+            int[] maxBlendshapes = new int[3] { blendshapeIDs.Length, blendshapeWeights.Length, blendshapeMinWeights.Length };
+            _maxBlendshapes = Mathf.Min(maxBlendshapes);
+
+        }
+
+        void Update()
+        {
+            if (audioLink.AudioDataIsAvailable())
+            {
+                // Convert to grayscale
+                float amplitude = Vector3.Dot(audioLink.GetDataAtPixel(delay, band), new Vector3(0.299f, 0.587f, 0.114f)) * 100f;
```

Comment:
Hmm this grayscale conversion looks sus to me. In the 4band section, each channel means a different thing (they are different levels of filtering, not chroma values). You should probably just select 1 single channel, and maybe put the channel select on a dropdown

Reply:
Set it to access the pixel's red value for now,
I like the idea of having an option to select which channel to read from a pixel, though that may be better as a enum in a different file